app/services/fetch_link_card_service.rb

```diff
@@ -11,7 +11,7 @@ class FetchLinkCardService < BaseService
       (#{Twitter::TwitterText::Regex[:valid_domain]})                                                                           #   $4 Domain(s)
       (?::(#{Twitter::TwitterText::Regex[:valid_port_number]}))?                                                                #   $5 Port number (optional)
       (/#{Twitter::TwitterText::Regex[:valid_url_path]}*)?                                                                      #   $6 URL Path and anchor
-      (\?#{Twitter::TwitterText::Regex[:valid_url_query_chars]}*#{Twitter::TwitterText::Regex[:valid_url_query_ending_chars]})? #   $7 Query String
+      (\?(#{Twitter::TwitterText::Regex[:valid_url_query_chars]}|\^)*#{Twitter::TwitterText::Regex[:valid_url_query_ending_chars]})? #   $7 Query String
```

Comment:
We already redefine `REGEXEN[:valid_url_query_chars]` in `config/initializers/twitter_regex.rb` to accept a wider range of characters than `twitter-text` does, so it would make the most sense to add to it there

Reply:
Good point, I missed this!